src/Promise/HttpPromise.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Client\Promise;
+
+use Http\Client\Exception;
+use Http\Promise\Promise;
+use Psr\Http\Message\ResponseInterface;
+
+interface HttpPromise extends Promise
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @return HttpPromise A new resolved http promise with value of the executed callback (onFulfilled / onRejected).
```

Comment:
This is not possible. Think about PHP 7, this line should still be `public function then(callable $onFulfilled = null, callable $onRejected = null): Promise;`